pkg/k8s/service_cache.go

```diff
@@ -548,23 +548,27 @@ func (s *ServiceCache) mergeServiceUpdateLocked(service *serviceStore.ClusterSer
 		s.externalEndpoints[id] = externalEndpoints
 	}
 
-	scopedLog.Debugf("Updating backends to %+v", service.Backends)
-	backends := map[string]*Backend{}
-	for ipString, portConfig := range service.Backends {
-		backends[ipString] = &Backend{Ports: portConfig}
-	}
-	externalEndpoints.endpoints[service.Cluster] = &Endpoints{
-		Backends: backends,
+	if !service.Shared {
+		delete(externalEndpoints.endpoints, service.Cluster)
```

Comment:
we don't need to check if the current cluster is remote or local, as externalEndpoints should not have any local cluster endpoint anyway.